### PR TITLE
[docs] EAS Build - new eas.json format

### PR DESCRIPTION
**Pull request by wkozyra95**
# Why

new eas.json format

# How

# Test Plan

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.io and README.md).
- [ ] This diff will work correctly for `expo build` (eg: updated `@expo/xdl`).
- [ ] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: docs/pages/build/eas-json.md [hunk 1]
```diff
@@ -13,122 +14,118 @@ import iosSchema from '~/scripts/schemas/unversioned/eas-json-ios-schema.js';
...
+      "node": "12.13.0",
+      "yarn": "1.22.5",
+      "env": {
+        "EXAMPLE_ENV": "example value"
+      },
+      "android": {
         "image": "default",
         "env": {
-          "EXAMPLE_ENV": "example value"
+            "PLATFORM": "android"
```
Comment (dsokal) [inline docs/pages/build/eas-json.md:91]:
```suggestion
          "PLATFORM": "android"
```

File: docs/pages/clients/eas-build.md [hunk 2]
```diff
@@ -19,32 +19,19 @@ If you have already have an `eas.json` file in your project, you'll need to upda
...
-      },
-      "development": {
-        "distribution": "internal",
-        "buildType": "development-client"
-      }
+  "build": {
+    "release": {},
+    "development": {
+      "distribution": "internal",
+      "developmentClient": true,
```
Comment (dsokal) [inline docs/pages/clients/eas-build.md:32]:
```suggestion
      "developmentClient": true
```

File: docs/pages/clients/eas-build.md [hunk 3]
```diff
@@ -19,32 +19,19 @@ If you have already have an `eas.json` file in your project, you'll need to upda
...
     }
-  }
+  },
```
Comment (dsokal) [inline docs/pages/clients/eas-build.md:34]:
```suggestion
  }
```

**Review by dsokal — APPROVED** [review]

File: docs/pages/build/eas-json.md [hunk 4]
```diff
@@ -213,12 +190,14 @@ where `PLATFORM_NAME` is one of `android` or `ios`.
 
 </details>
 
+## Common options for both paltforms
```
Comment (brentvatne) [inline docs/pages/build/eas-json.md:193]:
```suggestion
## Common options for both platforms
```

File: docs/pages/clients/eas-build.md [hunk 5]
```diff
@@ -19,30 +19,17 @@ If you have already have an `eas.json` file in your project, you'll need to upda
 
 <Tab >
 
-`expo-dev-client` does not modify your application's behavior in a build you would submit to the Play Store or App Store, so you must specify `development-client` as your `buildType` to create a custom development client.
+`expo-dev-client` does not modify your application's behavior in a build you would submit to the Play Store or App Store, so you must specify `developmentClient: true` to create a custom development client.
```
Comment (brentvatne) [inline docs/pages/clients/eas-build.md:22]:
```suggestion
`expo-dev-client` does not modify your application's behavior in a build you would submit to the Play Store or App Store, so you must specify `"developmentClient": true` to create a custom development client.
```

**Review by brentvatne — APPROVED** [review]

File: docs/pages/build/internal-distribution.md [hunk 6]
```diff
@@ -23,22 +23,18 @@ Open up `eas.json` and add a new build profile for iOS and/or Android.
...
-      "preview-enterprise" /* @end */: {
+    /* @info add this profile only if you have an account with Apple Developer Enterprise Program membership */
+    "preview-enterprise" /* @end */: {
+      "distribution": "internal",
+      "ios": {
         /* @info valid values: universal, adhoc */
         "enterpriseProvisioning": "universal",
         /* @end */
-        "distribution": "internal",
+
```
[the remark this answers is not in the document]
Comment (wkozyra95) [reply docs/pages/build/internal-distribution.md:37]: it has to empty line there, otherwise, indentation is broken

**Review by SaintPatricksCoin — APPROVED** [review]

**Review by SaintPatricksCoin — COMMENTED** [review]
1619798549686_1954324047_ca5176a7 gif
@SaintPatricksCoin

**Pull request merged**